addons/stock/report/report_stock_rule.py

```diff
@@ -11,6 +11,11 @@ class ReportStockRule(models.AbstractModel):
 
     @api.model
     def _get_report_values(self, docids, data=None):
+        if not data.get('product_id'):
+            data['product_id'] = docids or []
+        if not data.get('warehouse_ids'):
+            data['warehouse_ids'] = []
+
         product = self.env['product.product'].browse(data['product_id'])
         warehouses = self.env['stock.warehouse'].browse(data['warehouse_ids'])
```

Comment:
```suggestion
        product = self.env['product.product'].browse(data.get('product_id', docids))
        warehouses = self.env['stock.warehouse'].browse(data.get('warehouse_ids'))
```
is more pythonic

Reply:
Thanks. The thing is `data['product_id']` is used at multiple place in the code, do I change it everywhere?

Reply:
I didn't see those. As it's only there you can use `.get()` one more time yes. Thanks